### PR TITLE
[Security Content] Add Investigation Guides to Windows Rules - 2

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Adds IGs to the following rules:

- Privileged Account Brute Force
- Multiple Logon Failure Followed by Logon Success
- Multiple Logon Failure from the same Source Address
- Microsoft Build Engine Using an Alternate Name
- IIS HTTP Logging Disabled
- Renamed AutoIt Scripts Interpreter
- Potential Windows Error Manager Masquerading
- Process Activity via Compiled HTML File
- Registry Persistence via AppInit DLL
- Privilege Escalation via Named Pipe Impersonation

### CODE REVIEW

**Review by Aegrah — APPROVED** [review]
LGTM after addressing the unit testing error.

**Comment by w0rk3r** [comment]
@Aegrah The test will fail until https://github.com/elastic/detection-rules/pull/2513 merges

**Review by terrancedejesus — APPROVED** [review]
Content looks good!

Unit tests failing because of previous change to unit testing for OSquery which is required verbiage in the investigation guide.

Reference: https://github.com/elastic/detection-rules/blob/ceeb738c2917233171887f4aaec8b1b8f24cc349/tests/test_all_rules.py#L810

Screenshot 2023-02-22 at 12 46 11 PM

File: rules/windows/defense_evasion_execution_msbuild_started_renamed.toml [hunk 1]
```diff
@@ -18,14 +18,61 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+The Microsoft Build Engine is a platform for building applications. This engine, also known as MSBuild, provides an XML schema for a project file that controls how the build platform processes and builds software, and can be abused to proxy execution of code.
+
+This rule checks for renamed instances of MSBuild, which can indicate an attempt of evading detections, application allowlists, and other security protections.
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Examine the host for derived artifacts that indicates suspicious activities:
```
Comment (joepeeples) [inline rules/windows/defense_evasion_execution_msbuild_started_renamed.toml:36]:
```suggestion
- Examine the host for derived artifacts that indicate suspicious activities:
```

File: rules/windows/defense_evasion_masquerading_renamed_autoit.toml [hunk 2]
```diff
@@ -17,14 +17,61 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+AutoIt is a scripting language and tool for automating tasks on Microsoft Windows operating systems. Due to its capabilities, malicious threat actors can abuse it to create malicious scripts and distribute malware.
+
+This rule checks for renamed instances of AutoIt, which can indicate an attempt of evading detections, application allowlists, and other security protections.
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Examine the host for derived artifacts that indicates suspicious activities:
```
Comment (joepeeples) [inline rules/windows/defense_evasion_masquerading_renamed_autoit.toml:35]:
```suggestion
- Examine the host for derived artifacts that indicate suspicious activities:
```

File: rules/windows/defense_evasion_masquerading_werfault.toml [hunk 3]
```diff
@@ -19,6 +19,53 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
...
+By examining the specific traits of Windows binaries, such as process trees, command lines, network connections, registry modifications, etc., it's possible to establish a baseline of normal activity. Deviations from this baseline can indicate malicious activity, such as masquerading and deserve further investigation.
+
+This rule identifies a potential malicious process masquerading as `wermgr.exe` or `WerFault.exe`, by looking for a process creation with no arguments followed a by a network connection.
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Examine the host for derived artifacts that indicates suspicious activities:
```
Comment (joepeeples) [inline rules/windows/defense_evasion_masquerading_werfault.toml:35]:
```suggestion
- Examine the host for derived artifacts that indicate suspicious activities:
```

File: rules/windows/execution_via_compiled_html_file.toml [hunk 4]
```diff
@@ -25,14 +25,68 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+When users click those files, the HTML Help executable program (`hh.exe`) will execute them. `hh.exe` also can be used to execute code embedded in those files, PowerShell scripts, and executables, making it useful not only for attackers to proxy the execution of malicious payloads via a signed binary that could bypass security controls but to gain initial access to environments via social engineering methods.
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Investigate the parent process to gain understanding of what triggered this behavior.
+  - Retrieve `.chm`, `.ps1` and other files that were involved to further examination.
+- Examine the host for derived artifacts that indicates suspicious activities:
```
Comment (joepeeples) [inline rules/windows/execution_via_compiled_html_file.toml:43]:
```suggestion
- Examine the host for derived artifacts that indicate suspicious activities:
```

File: rules/windows/persistence_appinitdlls_registry.toml [hunk 5]
```diff
@@ -4,27 +4,79 @@ integration = ["endpoint", "windows"]
...
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Review the source process and related DLL file tied to the Windows Registry entry.
+  - Check whether the DLL is signed, and tied to a authorized program used on your environment.
+- Assess whether this behavior is prevalent in the environment by looking for similar occurrences across hosts.
+- Retrieve all DLLs under the AppInit registry keys:
+  - !{osquery{"query":"SELECT * FROM registry r where (r.key == 'HKEY_LOCAL_MACHINE\\\\SOFTWARE\\\\Microsoft\\\\Windows NT\\\\CurrentVersion\\\\Windows' or r.key == 'HKEY_LOCAL_MACHINE\\\\SOFTWARE\\\\Wow6432Node\\\\Microsoft\\\\Windows NT\\\\CurrentVersion\\\\Windows') and r.name == 'AppInit_DLLs'","label":"Osquery - Retrieve AppInit Registry Value"}}
+- Examine the host for derived artifacts that indicates suspicious activities:
```
Comment (joepeeples) [inline rules/windows/persistence_appinitdlls_registry.toml:40]:
```suggestion
- Examine the host for derived artifacts that indicate suspicious activities:
```

File: rules/windows/privilege_escalation_named_pipe_impersonation.toml [hunk 6]
```diff
@@ -17,17 +17,66 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
...
+
+Attackers can abuse named pipes to elevate their privileges by impersonating the security context in which they execute code. Metasploit, for example, creates a service and a random pipe, which uses the service to connect to the pipe, which impersonates the service security context, which is SYSTEM.
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Inspect the host for suspicious or abnormal behavior in the alert timeframe.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Examine the host for derived artifacts that indicates suspicious activities:
```
Comment (joepeeples) [inline rules/windows/privilege_escalation_named_pipe_impersonation.toml:34]:
```suggestion
- Examine the host for derived artifacts that indicate suspicious activities:
```

File: rules/windows/credential_access_bruteforce_admin_account.toml [hunk 7]
```diff
@@ -20,27 +20,45 @@ license = "Elastic License v2"
...
 
 ### Response and remediation
 
 - Initiate the incident response process based on the outcome of the triage.
-- If the host is a domain controller (DC):
-  - Activate your incident response plan for total Active Directory compromise.
-  - Review the privileges assigned to users that can access the DCs, to ensure that the least privilege principle is being followed and to reduce the attack surface.
-- Isolate the involved hosts to prevent further post-compromise behavior.
+- Isolate the source host to prevent further post-compromise behavior.
+- If the asset is exposed to the internet with RDP or other remote services available, take the necessary measures to restrict access to the asset by not exposing it preferentially. If not possible, limit the access via the firewall to only the needed IP addresses. Also, ensure the system uses robust authentication mechanisms and is patched regularly.
```
Comment (joepeeples) [inline rules/windows/credential_access_bruteforce_admin_account.toml:61]: I found "by not exposing it preferentially" hard to follow. Does that mean to just restrict/block access to the asset? Maybe the extra phrase is unnecessary then:
```suggestion
- If the asset is exposed to the internet with RDP or other remote services available, take the necessary measures to restrict access to the asset. If not possible, limit the access via the firewall to only the needed IP addresses. Also, ensure the system uses robust authentication mechanisms and is patched regularly.
```
But if the phrasing does add extra intended meaning, consider rephrasing to make a bit clearer.

File: rules/windows/credential_access_bruteforce_multiple_logon_failure_followed_by_success.toml [hunk 8]
```diff
@@ -20,27 +20,49 @@ license = "Elastic License v2"
...
 
 ### Response and remediation
 
 - Initiate the incident response process based on the outcome of the triage.
-- If the host is a domain controller (DC):
-  - Activate your incident response plan for total Active Directory compromise.
-  - Review the privileges assigned to users that can access the DCs, to ensure that the least privilege principle is being followed and to reduce the attack surface.
-- Isolate the involved hosts to prevent further post-compromise behavior.
+- Isolate the source host to prevent further post-compromise behavior.
+- If the asset is exposed to the internet with RDP or other remote services available, take the necessary measures to restrict access to the asset by not exposing it preferentially. If not possible, limit the access via the firewall to only the needed IP addresses. Also, ensure the system uses robust authentication mechanisms and is patched regularly.
```
Comment (joepeeples) [inline rules/windows/credential_access_bruteforce_multiple_logon_failure_followed_by_success.toml:65]: Same as above, "by not exposing it preferentially" is hard to follow. Remove or rephrase?

File: rules/windows/credential_access_bruteforce_multiple_logon_failure_same_srcip.toml [hunk 9]
```diff
@@ -41,10 +62,8 @@ note = """## Triage and analysis
 ### Response and remediation
 
 - Initiate the incident response process based on the outcome of the triage.
-- If the host is a domain controller (DC):
-  - Activate your incident response plan for total Active Directory compromise.
-  - Review the privileges assigned to users that can access the DCs, to ensure that the least privilege principle is being followed and to reduce the attack surface.
-- Isolate the involved hosts to prevent further post-compromise behavior.
+- Isolate the source host to prevent further post-compromise behavior.
+- If the asset is exposed to the internet with RDP or other remote services available, take the necessary measures to restrict access to the asset by not exposing it preferentially. If not possible, limit the access via the firewall to only the needed IP addresses. Also, ensure the system uses robust authentication mechanisms and is patched regularly.
```
Comment (joepeeples) [inline rules/windows/credential_access_bruteforce_multiple_logon_failure_same_srcip.toml:66]: Same as above, "by not exposing it preferentially" is hard to follow. Remove or rephrase?

File: rules/windows/defense_evasion_iis_httplogging_disabled.toml [hunk 10]
```diff
@@ -18,14 +18,48 @@ language = "eql"
...
 max_signals = 33
 name = "IIS HTTP Logging Disabled"
-note = """## Setup
+note = """## Triage and analysis
+
+### Investigating IIS HTTP Logging Disabled
+
+IIS (Internet Information Services) is a Microsoft web server software used to host websites and web applications on Windows. It provides features for serving dynamic and static content, and can be managed through a graphical interface or command-line tools.
+
+IIS logging is a data source that can be used for security monitoring, forensics, and incident response. It contains mainly information related to requests done to the webserver, and can be used to spot malicious activities like webshells. Adversaries can tamper, clear, and delete this data to evade detection, cover their tracks, and slow down incident response.
```
Comment (joepeeples) [inline rules/windows/defense_evasion_iis_httplogging_disabled.toml:27]:
```suggestion
IIS logging is a data source that can be used for security monitoring, forensics, and incident response. It contains mainly information related to requests done to the web server, and can be used to spot malicious activities like webshells. Adversaries can tamper, clear, and delete this data to evade detection, cover their tracks, and slow down incident response.
```

File: rules/windows/defense_evasion_masquerading_werfault.toml [hunk 11]
```diff
@@ -19,6 +19,53 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
 language = "eql"
 license = "Elastic License v2"
 name = "Potential Windows Error Manager Masquerading"
+note = """## Triage and analysis
+
+### Investigating Potential Windows Error Manager Masquerading
+
+By examining the specific traits of Windows binaries, such as process trees, command lines, network connections, registry modifications, etc., it's possible to establish a baseline of normal activity. Deviations from this baseline can indicate malicious activity, such as masquerading and deserve further investigation.
```
Comment (joepeeples) [inline rules/windows/defense_evasion_masquerading_werfault.toml:26]:
```suggestion
By examining the specific traits of Windows binaries -- such as process trees, command lines, network connections, registry modifications, and so on -- it's possible to establish a baseline of normal activity. Deviations from this baseline can indicate malicious activity, such as masquerading and deserve further investigation.
```

File: rules/windows/defense_evasion_masquerading_werfault.toml [hunk 12]
```diff
@@ -19,6 +19,53 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
...
+
+This rule identifies a potential malicious process masquerading as `wermgr.exe` or `WerFault.exe`, by looking for a process creation with no arguments followed a by a network connection.
```
Comment (joepeeples) [inline rules/windows/defense_evasion_masquerading_werfault.toml:28]:
```suggestion
This rule identifies a potential malicious process masquerading as `wermgr.exe` or `WerFault.exe`, by looking for a process creation with no arguments followed by a network connection.
```

File: rules/windows/defense_evasion_masquerading_werfault.toml [hunk 13]
```diff
@@ -19,6 +19,53 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
...
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Examine the host for derived artifacts that indicates suspicious activities:
+  - Analyze the process executable using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/defense_evasion_masquerading_werfault.toml:39]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/defense_evasion_execution_msbuild_started_renamed.toml [hunk 14]
```diff
@@ -18,14 +18,61 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+  - Analyze the process executable using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/defense_evasion_execution_msbuild_started_renamed.toml:40]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```
(I know this looks a little weird with all the s's -- an alternative might be `filtering by the 'process.entity_id' of the process` but that's a little more wordy.)

File: rules/windows/defense_evasion_masquerading_renamed_autoit.toml [hunk 15]
```diff
@@ -17,14 +17,61 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+  - Analyze the process executable using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/defense_evasion_masquerading_renamed_autoit.toml:39]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/execution_via_compiled_html_file.toml [hunk 16]
```diff
@@ -25,14 +25,68 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+  - Analyze the process executables, scripts and help files retrieved from the system using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/execution_via_compiled_html_file.toml:47]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/persistence_appinitdlls_registry.toml [hunk 17]
```diff
@@ -4,27 +4,79 @@ integration = ["endpoint", "windows"]
...
+  - Analyze the process executable and the DLLs using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/persistence_appinitdlls_registry.toml:44]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/privilege_escalation_named_pipe_impersonation.toml [hunk 18]
```diff
@@ -17,17 +17,66 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
...
+  - If any suspicious processes were found, examine the process executable using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/privilege_escalation_named_pipe_impersonation.toml:38]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/execution_via_compiled_html_file.toml [hunk 19]
```diff
@@ -25,14 +25,68 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
 license = "Elastic License v2"
 name = "Process Activity via Compiled HTML File"
-note = """## Setup
+note = """## Triage and analysis
+
+### Investigating Process Activity via Compiled HTML File
+
+CHM (Compiled HTML) files are a format for delivering online help files on Windows. CHM files are compressed compilations of various content, such as HTML documents, images, and scripting/web-related programming languages such as VBA, JScript, Java, and ActiveX.
+
+When users click those files, the HTML Help executable program (`hh.exe`) will execute them. `hh.exe` also can be used to execute code embedded in those files, PowerShell scripts, and executables, making it useful not only for attackers to proxy the execution of malicious payloads via a signed binary that could bypass security controls but to gain initial access to environments via social engineering methods.
```
Comment (joepeeples) [inline rules/windows/execution_via_compiled_html_file.toml:34]: Would "double-click" be more accurate? Also, trying to break up long sentence:
```suggestion
When users double-click CHM files, the HTML Help executable program (`hh.exe`) will execute them. `hh.exe` also can be used to execute code embedded in those files, PowerShell scripts, and executables. This makes it useful for attackers not only to proxy the execution of malicious payloads via a signed binary that could bypass security controls, but also to gain initial access to environments via social engineering methods.
```

File: rules/windows/execution_via_compiled_html_file.toml [hunk 20]
```diff
@@ -25,14 +25,68 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+
+#### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal behavior by the subject process such as network connections, registry or file modifications, and any spawned child processes.
+- Investigate the parent process to gain understanding of what triggered this behavior.
+  - Retrieve `.chm`, `.ps1` and other files that were involved to further examination.
```
Comment (joepeeples) [inline rules/windows/execution_via_compiled_html_file.toml:42]:
```suggestion
  - Retrieve `.chm`, `.ps1`, and other files that were involved to further examination.
```

File: rules/windows/persistence_appinitdlls_registry.toml [hunk 21]
```diff
@@ -4,27 +4,79 @@ integration = ["endpoint", "windows"]
...
+### Investigating Registry Persistence via AppInit DLL
+
+AppInit DLLs are dynamic-link libraries (DLLs) that are loaded into every process that creates a user interface (loads `user32.dll`) on Microsoft Windows operating systems. The AppInit DLL mechanism is used to load custom code into user-mode processes, allowing for the customization of the user interface and the behavior of Windows-based applications.
+
+Attackers who add those DLLs to the registry locations can execute code with elevated privileges, similar to process injection, and provide a solid and constant persistence on the machine.
+
+This rule identifies modifications on the AppInit registry keys.
+
+> **Note**:
+> This investigation guide uses the [Osquery Markdown Plugin](https://www.elastic.co/guide/en/security/master/invest-guide-run-osquery.html) introduced in Elastic Stack version 8.5.0. Older Elastic Stack versions will see unrendered Markdown in this guide.
```
Comment (joepeeples) [inline rules/windows/persistence_appinitdlls_registry.toml:30]:
```suggestion
> This investigation guide uses the [Osquery Markdown Plugin](https://www.elastic.co/guide/en/security/master/invest-guide-run-osquery.html) introduced in Elastic Stack version 8.5.0. Older Elastic Stack versions will display unrendered Markdown in this guide.
```

File: rules/windows/privilege_escalation_named_pipe_impersonation.toml [hunk 22]
```diff
@@ -17,17 +17,66 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
...
 license = "Elastic License v2"
 name = "Privilege Escalation via Named Pipe Impersonation"
-note = """## Setup
+note = """## Triage and analysis
+
+### Investigating Privilege Escalation via Named Pipe Impersonation
+
+A named pipe is a type of inter-process communication (IPC) mechanism used in operating systems like Windows, which allows two or more processes to communicate with each other by sending and receiving data through a well-known point.
+
+Attackers can abuse named pipes to elevate their privileges by impersonating the security context in which they execute code. Metasploit, for example, creates a service and a random pipe, which uses the service to connect to the pipe, which impersonates the service security context, which is SYSTEM.
```
Comment (joepeeples) [inline rules/windows/privilege_escalation_named_pipe_impersonation.toml:26]: Reducing repetition of "which" phrases -- please confirm whether this changes any of the description's meaning/accuracy:
```suggestion
Attackers can abuse named pipes to elevate their privileges by impersonating the security context in which they execute code. Metasploit, for example, creates a service and a random pipe, and then uses the service to connect to the pipe and impersonate the SYSTEM service security context.
```

**Review by joepeeples — COMMENTED** [review]
Great content as always, thanks for preparing it! Just added some nits here and there.

**Pull request merged**